src/components/ThirdPartyCollectionDetailPage/CollectionPublishButton/CollectionPublishButton.tsx

```diff
@@ -53,11 +53,30 @@ const CollectionPublishButton = (props: Props) => {
     onClick(collection.id, itemIds, buttonAction)
   }, [collection, items, buttonAction])
 
+  const itemsTryingToPublish = useMemo(
+    () => items.filter(item => !itemCurations.find(itemCuration => itemCuration.itemId === item.id)).length,
+    [items, itemCurations]
+  )
+
+  const underReviewButtonLabel = useMemo(() => {
+    switch (buttonAction) {
+      case PublishButtonAction.PUBLISH:
+        if (itemsTryingToPublish) {
+          return t('third_party_collection_detail_page.cant_publish_items', { count: itemsTryingToPublish })
+        }
+        return t('third_party_collection_detail_page.cant_publish')
+      case PublishButtonAction.PUBLISH_AND_PUSH_CHANGES:
+        return t('third_party_collection_detail_page.cant_publish_and_push_changes', { count: itemsTryingToPublish })
+      default:
+        return t('third_party_collection_detail_page.cant_publish')
+    }
+  }, [buttonAction, itemsTryingToPublish])
+
   const hasPendingItemCurations = itemCurations && !!itemCurations.find(ic => ic.status === CurationStatus.PENDING)
   const isTryingToPublish = [PublishButtonAction.PUBLISH, PublishButtonAction.PUBLISH_AND_PUSH_CHANGES].includes(buttonAction)
 
   return !isLoadingItemCurations && isTryingToPublish && hasPendingItemCurations ? (
-    <UnderReview type="publish" />
+    <UnderReview content={underReviewButtonLabel} />
```

Comment:
This works! but hear me out.

I feel like the original `UnderReview` component made sense because it avoided having to repeat the Popup logic and also decided which translation to show:

```ts
if (collection.isApproved) {
  if (hasPendingCuration) {
    button = <UnderReview type="push" />
  } else if (status === SyncStatus.UNSYNCED) {
    // (...)
  }
} else {
  button = <UnderReview type="publish" />
}
```

in our case, the component doesn't seem to be abstracting much, as we pass the entire text and just let it wrap a button with a popup.

So I was thinking, what about (options):
- Letting the component decide which translation to use by receiving: `itemsTryingToPublish` and `buttonAction`
- Pass `items` and `itemCurations` to the component and move the logic there (both memos)
- Delete underreview and move the code here
- Some combination of the three

I don't know if any of those is better than the others or even than the current solution. There might be other options I haven't considered. What do you think?

Reply:
Agreed! I think we don't need an extra component in this case, so I went with option 3 :) 
`UnderReview` component deleted and `PopUp` included in the jsx from the `CollectionPublishButton`